spec/lib/miq_automation_engine/service_methods/miq_ae_service_manageiq-providers-redhat-infra_manager-vm_spec.rb

```diff
@@ -0,0 +1,28 @@
+module MiqAeServiceManageIQ_Providers_Redhat_InfraManager_VmSpec
+  describe MiqAeMethodService::MiqAeServiceManageIQ_Providers_Redhat_InfraManager_Vm do
+    let(:vm)         { FactoryGirl.create(:vm_redhat) }
+    let(:service_vm) { MiqAeMethodService::MiqAeServiceManageIQ_Providers_Redhat_InfraManager_Vm.find(vm.id) }
+
+    before do
+      allow_any_instance_of(Vm).to receive(:my_zone).and_return('default')
```

Comment:
Can you get rid of `allow_any_instance_of` and only mock `vm` and `service_vm` if needed?